pkg/v21/chartconfig/current.go

```diff
@@ -36,13 +37,20 @@ func (c *ChartConfig) GetCurrentState(ctx context.Context, clusterConfig Cluster
 		LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, project.Name()),
 	}
 
+	ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
+	defer cancel()
+
 	chartConfigList, err := cc.Client.TenantCluster.G8s.CoreV1alpha1().ChartConfigs(resourceNamespace).List(listOptions)
 	if tenant.IsAPINotAvailable(err) {
 		c.logger.LogCtx(ctx, "level", "debug", "message", "tenant cluster is not available yet")
 		c.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
 		resourcecanceledcontext.SetCanceled(ctx)
 		return nil, nil
-
+	} else if ctx.Err() == context.DeadlineExceeded {
```

Comment:
I think this `if` should go after the error handling. It will wait for the `CoreV1alpha1().ChartConfigs()` to return anyway. If it errors we should know about it.

Reply:
Agreed and good point. There might be more cases where we need to handle the error and cancel the resource but we should know about them.

I updated the migration resources as well.